data/mods/ssb/abilities.ts

```diff
@@ -25,6 +25,37 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 			}
 		},
 	},
+  // dream
+	greedpunisher: {
+		desc: 'This pokemon can only be damaged by direct attacks. On switch-in, boosts stats based on hazards on the field. 1 stat raised if one-two hazards are up, 2 stats raised if two or more are up.',
+		shortDesc: "On switch-in, boosts stats based on hazards present on the user's side.",
+		name: 'Greed Punisher',
```

Comment:
```suggestion
		desc: "This Pokemon can only be damaged by direct attacks. On switch-in, this Pokemon's stats are boosted based on the number of hazards on the field. 1 stat is raised if 1-2 hazards are up, and 2 stats are raised if 3 or more hazards are up.",
		shortDesc: "On switch-in, boosts stats based on the number of hazards up on this Pokemon's side.",
		name: "Greed Punisher",
```
check that the new shortdesc works